packages/shared/src/authz/policies/canCreateWorkspaceProject.ts

```diff
@@ -0,0 +1,103 @@
+import { AuthPolicy } from '../domain/policies.js'
+import {
+  ServerNoAccessError,
+  ServerNoSessionError,
+  WorkspaceLimitsReachedError,
+  WorkspaceNoAccessError,
+  WorkspaceNoEditorSeatError,
+  WorkspaceNotEnoughPermissionsError,
+  WorkspacesNotEnabledError,
+  WorkspaceSsoSessionNoAccessError
+} from '../domain/authErrors.js'
+import { err, ok } from 'true-myth/result'
+import { hasMinimumServerRole } from '../checks/serverRole.js'
+import { Roles } from '../../core/constants.js'
+import { maybeMemberRoleWithValidSsoSessionIfNeeded } from '../fragments/workspaceSso.js'
+import { hasAnyWorkspaceRole } from '../checks/workspaceRole.js'
+import { throwUncoveredError } from '../../core/index.js'
+import { hasEditorSeat } from '../checks/workspaceSeat.js'
+import { MaybeUserContext, WorkspaceContext } from '../domain/context.js'
+
+export const canCreateWorkspaceProjectPolicy: AuthPolicy<
+  | 'getEnv'
```

Comment:
I'm wondering if this should just be `canCreateProjectPolicy` instead, and then inside if it we differentiate between whether it's a workspace or non-workspace policy? Would simplify the permission API a bit

Reply:
I have the non-workspace one WIP, will see how much they overlap after. (The non-workspace one is stupid simple compared to this.)